### PR TITLE
My Site Dashboard: Initial Screen App Setting

### PR DESCRIPTION
**Pull request by zwarm**
Parent #15989 

This PR adds the “initial screen” setting to the App Settings view. The setting is only visible when tabs are enabled for My Site Dashboard. This setting will persist until the app is uninstalled/reinstalled - similar to other device app settings.

**Merge Instructions**
- You can review, but not merge this PR yet. I am waiting on some verifications from the iOS team with regards to what the default value should be for the setting. I will review the "do not merge label" after the direction has been clarified.

**To test**
Do a clean install of the app and don't enable the MySiteDashboardTabsFeatureConfig flag. Then:

1. Login or create a new account
2. Navigate to Me → App Settings 
3. Note: You don’t see an "Initial Screen" option 
4. Navigate to Me → App Settings → Debug
5. Enable the MySiteDashboardTabsFeatureConfig flag and restart the App
6. When the app has restarted, Navigate to Me → App Settings 
7. Note: You see an "Initial Screen" option
8. Change the value
9. Make sure that an event is fired on the console, like: `🔵 Tracked: app_settings_initial_screen_changed, Properties: {"default_tab_experiment":"site_menu","selected":"site_menu"}`
10. Close and reopen the app
11. Navigate back to Me → App Settings
12. Note: The option you selected is now the default

## Regression Notes
1. Potential unintended areas of impact
The app setting does not show or hide the initial screen setting

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual test

3. What automated tests I added (or what prevented me from doing so)
There are no unit tests for the AppSettingsFragment.

PR submission checklist:
- [X] I have completed the Regression Notes.
- [X] I have considered adding accessibility improvements for my changes.
- [X] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -4005,4 +4005,20 @@ translators: %s: Select control option value e.g: "Auto, 25%". -->
     <string name="blogging_prompts_onboarding_try_it_now">Try it now</string>
     <string name="blogging_prompts_onboarding_remind_me">Remind me</string>
     <string name="blogging_prompts_onboarding_card_prompt">Cast the movie of your life.</string>
+
+    <!-- Initial Screen -->
+    <string name="initial_screen">Initial Screen</string>
+    <string name="initial_screen_home">Home</string>
+    <string name="initial_screen_site_menu">Site Menu</string>
```
Comment (ashiagr) [inline WordPress/src/main/res/values/strings.xml]: Nitpick: I wonder if we can point to the below strings with `translatable=false` here as this initial screen setting directly relates to these tabs and their content. If we change tab titles in the future then changes will be reflected here as well. Wdyt?

```
<string name="my_site_dashboard_tab_title">Home</string>
<string name="my_site_menu_tab_title">Site Menu</string>
```

**Review by ashiagr — COMMENTED** [review]
LGTM so far, @zwarm. 👍 I just left one comment. I'll wait to review till further implementation.

File: WordPress/src/main/res/values/strings.xml (hunk 1 above)
Comment (zwarm) [reply WordPress/src/main/res/values/strings.xml]: > Nitpick: I wonder if we can point to the below strings with `translatable=false` here as this initial screen setting directly relates to these tabs and their content. If we change tab titles in the future then changes will be reflected here as well. Wdyt?
> 
> ```
> <string name="my_site_dashboard_tab_title">Home</string>
> <string name="my_site_menu_tab_title">Site Menu</string>
> ```

Yes 👍  Perfect idea!!! I'll make the change.

Comment (zwarm) [reply WordPress/src/main/res/values/strings.xml]: @ashiagr - I have updated the PR to include the "string" suggestion above and set the default value to site_menu for the settings. This PR is ready for review and merge (after CI). My next PR will update the implementation of the "default" tab + experiment to use this pref. It should make things easier for the default tab logic, as we'll just need to use the initial value. 🤞

Comment (ashiagr) [reply WordPress/src/main/res/values/strings.xml]: Got it! I'll hit the merge button now.

**Review by ashiagr — APPROVED** [review]
LGTM 👍 
Feel free to merge it when CI is done.

**Pull request merged**